helpers/auth_helper.py

```diff
@@ -0,0 +1,29 @@
+"""An auth_helper (Singleton) is created in the module, which will be used
+    in the application to authenticate users.
+"""
+
+from flask_httpauth import HTTPBasicAuth
+from werkzeug.security import check_password_hash
+
+
+class HTTPAuthHelper():
+    """User authentication helper class(Singleton)."""
+    def __init__(self):
+        self.auth = HTTPBasicAuth()
+        self.auth.verify_password(self.verify_password)
+        self.credentials = {}
```

Comment:
Is not this good case for inheritance? So it'll be used as `@auth_helper.login_required` instead of `@auth_helper.auth.login_required` that looks quite natural.

BTW, I would name it `HTTPAuthProvider` instead of `HTTPAuthHelper`. 'Helper' is a kind of second class citizen in the code. `@auth_provider.login_required` would look nice.

Reply:
Now inheritance is used. `auth_helper` renamed to `auth_provider`. But I saved its location in `helpers`, because IMHO it's just an object that helps authenticate users, but authentication is not the target of the service.